akka-cluster-tools/src/main/scala/akka/cluster/pubsub/DistributedPubSubMediator.scala

```diff
@@ -251,6 +238,36 @@ object DistributedPubSubMediator {
     @SerialVersionUID(1L)
     final case class SendToOneSubscriber(msg: Any)
 
+    /*
+     * Messages used to encode protocol to make sure that we do not send Subscribe/Unsubscribe message to 
+     * child (mediator -> topic, topic -> group) during a period of transition. Protects from situations like:
+     * 
+     * Sending Subscribe/Unsubscribe message to child actor after child has been terminated 
+     * but Terminate message did not yet arrive to parent.
+     *
+     * Sending Subscribe/Unsubscribe message to child actor that has Prune message queued and pruneDeadline set.
+     *
+     * In both of those situation parent actor still thinks that child actor is alive and forwards messages to it resulting in lost ACKs.
+     */
+    /**
```

Comment:
My intention was to describe why NoMoreSubscribers, TerminateRequest, NewSubscriberArrived were added in the first place, putting this comment inside java/scaladoc would seem like it only describes this particular message.

Maybe those three messages could be enclosed in some external object and this comment could be included in java/scaladoc for this enclosing object.